docs/src/main/asciidoc/security-oidc-auth0-tutorial.adoc

```diff
@@ -0,0 +1,778 @@
+====
+
+Configure OIDC in the Quarkus `application.properties` file  by using the properties from the Auth0 application that you created earlier.
+
+[source,configuration]
+----
+# Make sure the application domain is prefixed with 'https://'
+quarkus.oidc.auth-server-url=https://dev-3ve0cgn7.us.auth0.com
+quarkus.oidc.application-type=web-app
+quarkus.oidc.client-id=sKQu1dXjHB6r0sra0Y1YCqBZKWXqCkly
+quarkus.oidc.credentials.secret=${client-secret}
+----
+
+In completing this step, you have just configured Quarkus to use the domain, client ID, and secret of yourAuth0 application.
+Setting the property `quarkus.oidc.application-type=web-app` instructs Quarkus to use the OIDC authorization code flow, but there are also other methods, which are discussed later on in the tutorial.
+
+The endpoint address will be  \http://localhost:8080/hello, which must also be registered as a permitted callback URL in your Auth0 application.
+
+image::auth0-allowed-callback.png[Auth0 allowed callback URL]
+
+After completing this step, when you access the Quarkus \http://localhost:8080/hello endpoint from a browser, Auth0 redirects you back to the same address after the authentication is completed.
+
+[NOTE]
+====
+By default, Quarkus automatically uses the current request path as the callback path.
+But you can override the default behavior and configure a specific callback path by setting the Quarkus `quarkus.oidc.authentication.redirect-path` property, as outlined in the following example:
+
+`quarkus.oidc.authentication.redirect-path=/hello`
+
+In production, your application will most likely have a larger URL space, with many more endpoint addresses available. In such cases you will configure Quarkus with a dedicated callback (redirect) path, for example, `quarkus.oidc.authentication.redirect-path=/authenticated-welcome` and register this URL in the provider's dashboard as `Auth0` will not allow an open ended list of redirect URLs. Quarkus will call `/authenticated-welcome` after accepting a redirect from `Auth0` and completing the Authorization code flow and creating the session cookie - and now your application can let authenticated users access other parts of the secured application space, for example, by using JAX-RS API to redirect, without requiring users to reauthenticate.
+====
+
+Now you are ready to start testing the endpoint.
+
+== Test Quarkus endpoint - first try
+
+Start Quarkus in `devmode`:
+
+[source,bash]
+----
+$ mvn quarkus:dev
+----
+
+This is the only time you are expected to start Quarkus manually in devmode while working with this tutorial. All the configuration and code updates in the follow up sections is expected to be recognized by Quarkus without having to restart the application manually.
+
+Open the browser and access `http://localhost:8080/hello`, you will be redirected to `Auth0` where you will login:
+
+image::auth0-login.png[Auth0 Login]
+
+and authorize the `QuarkusAuth0` application to access your account:
```

Comment:
```suggestion
Next, give the `QuarkusAuth0` application authorization to access your account:
```

Reply:
@michelle-purcell It reads a little bit unusual in scope of describing the OIDC code flow, though it is surely a perfect text as is :-), I'll keep the original text if you don't mind - this is a typical combination AFAIK, `user authorizes` or `user grants permissions`

Reply:
:+ - One thing: fix capitalization of the sentence. How about:

```suggestion
Authorize the `QuarkusAuth0` application to access your account:
```